test/functional/legacy/094_visual_mode_operators_spec.lua

```diff
@@ -0,0 +1,161 @@
+-- Test for Visual mode and operators.
+--
+-- Tests for the two kinds of operations: Those executed with Visual mode
+-- followed by an operator and those executed via Operator-pending mode. Also
+-- part of the test are mappings, counts, and repetition with the . command.
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+-- Vim script user functions needed for some of the test cases.
+local function source_user_functions()
+  source([[
+    function MoveToCap()
+      call search('\u', 'W')
+    endfunction
+    function SelectInCaps()
+      let [line1, col1] = searchpos('\u', 'bcnW')
+      let [line2, col2] = searchpos('.\u', 'nW')
+      call setpos("'<", [0, line1, col1, 0])
+      call setpos("'>", [0, line2, col2, 0])
+      normal! gv
+    endfunction
+  ]])
+end
+
+describe('Visual mode and operator', function()
+  before_each(function()
+    clear()
+    source_user_functions()
+  end)
+
+  describe('simple change in Visual mode', function()
```

Comment:
since you aren't using setup(), and the describe() blocks don't contain multiple it() specs, the nested describe()s are not needed. Sorry for the confusion; I thought you had a different setup() call for some of the describe()s previously.